packages/babel-core/src/transformation/normalize-file.ts

```diff
@@ -59,42 +62,105 @@ export default function* normalizeFile(
     }
 
     if (!inputMap) {
-      const lastComment = extractComments(INLINE_SOURCEMAP_REGEX, ast);
-      if (lastComment) {
-        try {
-          inputMap = convertSourceMap.fromComment("//" + lastComment);
-        } catch (err) {
-          if (process.env.BABEL_8_BREAKING) {
-            console.warn(
-              "discarding unknown inline input sourcemap",
-              options.filename,
-              err,
-            );
-          } else {
-            debug("discarding unknown inline input sourcemap");
+      if (process.env.BABEL_8_BREAKING || !fromAst) {
```

Comment:
Discussion welcome here.
I'm not sure if source map comments are included in `program.comments` when `transformFromAst`, and maybe we shouldn't use the fast path in Babel 8 either.